spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/config/RabbitServiceAutoConfiguration.java

```diff
@@ -88,27 +90,36 @@ public Cloud cloud() {
 			protected static class UseCloudConnectors {
 
 				/**
-				 * Creates a {@link ConnectionFactory} using the singleton service
-				 * connector.
-				 *
+				 * Creates a {@link ConnectionFactory} using the singleton service connector.
 				 * @param cloud {@link Cloud} instance to be used for accessing services.
+				 * @param connectorConfigObjectProvider the {@link ObjectProvider} for the
+				 * {@link ServiceConnectorConfig}; in this case the expectation is about
+				 * {@code org.springframework.cloud.service.messaging.RabbitConnectionFactoryConfig}.
 				 * @return the {@link ConnectionFactory} used by the binder.
 				 */
 				@Bean
 				@Primary
-				ConnectionFactory rabbitConnectionFactory(Cloud cloud) {
-					return cloud.getSingletonServiceConnector(ConnectionFactory.class, null);
+				ConnectionFactory rabbitConnectionFactory(Cloud cloud,
+						ObjectProvider<ServiceConnectorConfig> connectorConfigObjectProvider) {
```

Comment:
Don't we need to explicitly look for a `RabbitConnectionFactoryConfig` ?

Otherwise, we might attempt to configure the connection factory with a different set of properties which may cause undesired side-effects.

Also, what happens if there are multiple `ServiceConnectorConfig` beans?